ironfish/src/network/peerNetwork.test.ts

```diff
@@ -534,6 +542,109 @@ describe('PeerNetwork', () => {
+        const block8 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block8)
+        const block9 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block9)
+        const block10 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block10)
+
+        const { peer } = getConnectedPeer(peerNetwork.peerManager)
+        const peerIdentity = peer.getIdentityOrThrow()
+
+        const sendSpy = jest.spyOn(peer, 'send')
+
+        const rpcId = 432
+        const message = new GetBlockHeadersRequest(block2.header.sequence, 4, 6, false, rpcId)
+        // The start header, block2, is the first lookup. Then we skip 6,
+        // resulting in 7 lookups. The 8th and final lookup is block9.
+        const response = new GetBlockHeadersResponse([block2.header, block9.header], rpcId)
+
+        await peerNetwork.peerManager.onMessage.emitAsync(peer, { peerIdentity, message })
+
+        expect(sendSpy.mock.calls[0][0]).toBeInstanceOf(GetBlockHeadersResponse)
+        expectGetBlockHeadersResponseToMatch(
+          sendSpy.mock.calls[0][0] as GetBlockHeadersResponse,
+          response,
+        )
+      })
+
+      it('does not include headers beyond the limit', async () => {
+        const { node, peerNetwork } = nodeTest
+        const block2 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block2)
+        const block3 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block3)
+        const block4 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block4)
+        const block5 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block5)
+        const block6 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block6)
+        const block7 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block7)
+        const block8 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block8)
+        const block9 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block9)
+        const block10 = await useMinerBlockFixture(node.chain)
+        await expect(node.chain).toAddBlock(block10)
+
+        const { peer } = getConnectedPeer(peerNetwork.peerManager)
+        const peerIdentity = peer.getIdentityOrThrow()
+
+        const sendSpy = jest.spyOn(peer, 'send')
+
+        const rpcId = 432
+        const message = new GetBlockHeadersRequest(block2.header.sequence, 4, 7, false, rpcId)
+        // The start header, block2, is the first lookup. Then we skip 7,
+        // resulting in 8 lookups. Thus, the header from block10 is not included
+        // since it would be the 9th lookup.
+        const response = new GetBlockHeadersResponse([block2.header], rpcId)
```

Comment:
I'm not the biggest fan of these really fiddly tests that require comments to clearly understand what it is testing, but not sure of a better alternative

Reply:
The only change I'd want to make is maybe move the mocking of MAX_REQUESTED_HEADERS closer to the actual test. I guess also it's tricky because it mocks MAX_REQUESTED_HEADERS, but it's actually testing that maxLookups is maxheaders * 2, whereas maybe a better test would set a value for maxLookups then test that you don't go over it.